### PR TITLE
Compute `compute_rep_shape` for DpasLayout

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

File: python/test/unit/language/test_core.py [hunk 1]
```diff
@@ -4779,6 +4779,11 @@ def compute_rep_shape(layout):
         warp_shape = np.multiply(layout.sz_per_thread, layout.threads_per_warp)
         rep_shape = np.multiply(warp_shape, layout.warps_per_cta)
         return rep_shape
+    elif type(layout) is DpasLayout:
+        size_per_thread = [layout.repeatCount * layout.execution_size / layout.threads_per_warp, 1]
+        warp_shape = np.multiply(size_per_thread, layout.threads_per_warp)
```
Comment (chengjunlu) [inline python/test/unit/language/test_core.py:4784]: The warp_shape should be [layout.repeatCount,  layout.execution_size] I think.

File: python/test/unit/language/test_core.py [hunk 2]
```diff
@@ -4779,6 +4779,11 @@ def compute_rep_shape(layout):
...
+        rep_shape = np.multiply(warp_shape, layout.warps_per_cta)
```
Comment (chengjunlu) [inline python/test/unit/language/test_core.py:4784]: The logic seems aligned to the BlockLayout.
The expression seems to compute the size of `ShapePerCTA`
```
/* The difference between ShapePerCTATile and ShapePerCTA:
 * (1) ShapePerCTATile is defined by SizePerThread * ThreadsPerWarp *
 *     WarpsPerCTA in each dimension and is independent from the tensor shape.
 * (2) ShapePerCTA is defined by shape / CTASplitNum in each dimension.
 * (3) In the implementation of emitIndices, ShapePerCTATile will
 *     be replicated or wrapped to fit ShapePerCTA.
 */
```

But the name `rep` is confusing. If the `rep` represent for the `replicate`, it is not aligned.

File: python/test/unit/language/test_core.py (hunk 1 above)
Comment (whitneywhtsang) [reply python/test/unit/language/test_core.py]: done.

File: python/test/unit/language/test_core.py (hunk 2 above)
Comment (whitneywhtsang) [reply python/test/unit/language/test_core.py:4784]: I also think it is calculating `ShapePerCTATile`.

**Review by alexbaden — APPROVED** [review]
Resolves issue w/ #906 for me

**Pull request merged**